core/dbt/contracts/graph/nodes.py

```diff
@@ -552,6 +540,30 @@ def depends_on_macros(self):
         return self.depends_on.macros
 
 
+@dataclass
+class FileSlice(dbtClassMixin, Replaceable):
```

Comment:
I'm not a fan of including the "Replaceable" class in every single class we create and in general think that it's a bad pattern anyway. Can we leave it off and if somebody really wants that functionality on a particular object it can be specifically added?

Reply:
I will leave it out, with pleasure!

Reply:
This was me by the way 🙈 I believe I included it because we included it on everything 🙃 I'll make a note for myself that it's not a pattern we want to continue.

Reply:
Yeah, I wouldn't mind a tech debt ticket to remove it from every place that it's not needed, and look at the places where it's actually called to see if it should be. Some of that was due to having Compiled and non-compiled versions of nodes, I think.

Reply:
https://github.com/dbt-labs/dbt-core/issues/7802